test/test_transforms.py

```diff
@@ -1685,6 +1685,67 @@ def test_random_erasing(self):
             img_re = random_erasing(img)
             self.assertTrue(torch.equal(img_re, img))
 
+    def test_compose_non_scriptable(self):
+        # T.Compose is now nn.Module, we should check BC with any exotic ops inside
+        # like Lambda, custom function etc
+        img_np = np.random.randint(0, 256, (34, 42, 3), np.uint8)
+        img_pil = Image.fromarray(img_np, mode='RGB')
+
+        def custom_transform(img):
+            return Image.blend(img, img, alpha=0.5)
+
+        t = transforms.Compose([
+            transforms.RandomResizedCrop(15),
+            transforms.Lambda(lambda x: x),
+            custom_transform,
+        ])
+
+        t.transforms.append(transforms.ToTensor())
+        t.transforms.append(lambda x: x + 2)  # THIS DOES NOT WORK
+        t.transforms.extend([
+            lambda x: x ** 2,  # THIS DOES NOT WORK
+            transforms.Normalize((0.485, 0.456, 0.406), (0.229, 0.224, 0.225)),
+        ])
+
+        res = t(img_pil)
+        self.assertTrue(isinstance(res, torch.Tensor))
+
+    def test_random_apply_non_scriptable(self):
+        # T.RandomApply is now nn.Module, we should check BC with any exotic ops inside
+        # like Lambda, custom function etc
+        img_np = np.random.randint(0, 256, (34, 42, 3), np.uint8)
+        img_pil = Image.fromarray(img_np, mode='RGB')
+
+        def custom_transform(img):
+            return Image.blend(img, img, alpha=0.5)
+
+        t = transforms.RandomApply([
+            transforms.RandomResizedCrop(15),
+            lambda x: x,
+            transforms.Lambda(lambda x: x),
+            custom_transform,
+        ])
+
+        res = t(img_pil)
+        self.assertTrue(isinstance(res, Image.Image))
+
+    def test_lambda_attributes(self):
+
+        class MyTransform:
+
+            def __init__(self, s):
+                self.s = s
+
+            def __call__(self, x):
+                return x + self.s
+
+        t = MyTransform(10)
+        t_lambda = transforms.Lambda(t)
+        self.assertEqual(t_lambda.s, t.s)
+        t_lambda.s = 5
+        self.assertEqual(t_lambda.s, t.s)
+        self.assertEqual(t_lambda(10), t(10))
```

Comment:
Hum, now I think I see why you override `__getattr__`. This wasn't the case before, I suppose this is to try to keep BC with Compose, if the user wants to access those fields after we wrapped they non-nn module in a `Lambda`?

My first thought would be to avoid this type of magic as it could be confusing to the user, but I'm not sure I see a better solution for this yet, apart from encouraging all new use-cases to use `nn.Sequential`, and letting `Compose` be not recommended module to use.

Thoughts?

Reply:
Yes, the reason to override `__getattr__` and `__setattr__` is to be BC with Compose. More details [above](https://github.com/pytorch/vision/pull/2645#discussion_r484452454).